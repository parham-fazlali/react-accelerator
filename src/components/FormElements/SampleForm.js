/* @flow */

import React, { PureComponent } from 'react';
import { compose } from 'redux';
import { Field, reduxForm } from 'redux-form';

import { isFill } from 'utils/validation';
import {
  InputCheckbox,
  InputDate,
  InputNumber,
  InputRadio,
  InputSelect,
  InputText,
  InputTime,
  InputTextArea,
  SubmitButton
} from 'components/FormElements';

type Props = {
  submitting: boolean,
  handleSubmit: Function
};

type State = {
  afterChangeState: boolean
};
const USERS_NEW_FORM = 'SampleForm';

class SampleForm extends PureComponent<Props, State> {
  state = {
    afterChangeState: false
  };

  genderOptions = [
    { title: 'Man', value: 'man' },
    { title: 'Woman', value: 'woman' }
  ];

  jobOptions = [
    { label: 'Designer', name: 'Designer', key: 'designer' },
    { label: 'Teacher', name: 'Teacher', key: 'teacher' },
    { label: 'Web Developer', name: 'Web Developer', key: 'web_developer' }
  ];

  validate = {
    isFill: isFill()
  };

  handleTextArea = text => {
    console.log('handleTextArea', text);
  };

  render() {
    const { submitting, handleSubmit } = this.props;
    const { afterChangeState } = this.state;
    return (
      <div>
        <form>
          <Field name="name" size="large" component={InputText} label="Name" />

          <Field
            name="gender"
            size="large"
            component={InputRadio}
            options={this.genderOptions}
            label="Gender"
          />

          <Field
            name="job"
            size="large"
            label="Job"
            component={InputSelect}
            options={this.jobOptions}
          />

          <Field
            name="childs"
            size="large"
            label="Childs"
            component={InputNumber}
          />

          <Field
            name="birthday"
            size="large"
            label="Birthday"
            component={InputDate}
          />
          {afterChangeState ? (
            <Field
              name="bio"
              label="Bio"
              component={InputTextArea}
              afterChange={() => {}}
              validate={this.validate.isFill}
            />
          ) : (
            <Field
              name="bio"
              label="Bio"
              component={InputTextArea}
              validate={this.validate.isFill}
            />
          )}

          <Field
            name="agreement"
            size="large"
            label="I agree the terms and conditions"
            component={InputCheckbox}
          />

          <Field
            name="agreement"
            size="large"
            label="I agree the terms and conditions"
            component={InputTime}
          />

          <SubmitButton
            label="Add user"
            size="large"
            type="primary"
            shape="round"
            icon="check"
            loading={submitting}
            disabled={submitting}
            onSubmit={handleSubmit}
          />
        </form>
      </div>
    );
  }
}

export default compose(
  reduxForm({
    form: USERS_NEW_FORM
  })
)(SampleForm);
