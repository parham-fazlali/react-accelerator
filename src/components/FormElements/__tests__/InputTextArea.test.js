import React from 'react';
import { mount } from 'enzyme';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';
import { get } from 'lodash';
import SampleForm from '../SampleForm';

global.Input = require('antd').Input;

describe('<SampleForm /> on InputTextArea', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = createStore(combineReducers({ form: formReducer }));
    wrapper = mount(
      <Provider store={store}>
        <SampleForm id="formelement" />
      </Provider>
    );
  });

  afterEach(() => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Test InputTextArea with isFill validation error', () => {
    const InputTextArea = wrapper.find('InputTextArea').instance();
    const input = wrapper.find('#bio');
    input.at(0).simulate('blur', { target: { value: '' } });
    expect(get(InputTextArea, 'props.meta.error', false)).toBe('Required');
    expect(get(InputTextArea, 'props.meta.touched', false)).toBeTruthy();
  });

  it('Test InputTextArea in change without afterChange props', () => {
    const input = wrapper.find('#bio');
    input.at(0).simulate('change', { target: { value: 'sample' } });
    expect(input.at(0).text()).toEqual('sample');
  });

  it('Test InputTextArea Change event with afterChange props', () => {
    const input = wrapper.find('SampleForm').instance();
    input.setState({ afterChangeState: true });
    const textArea = wrapper.find('#bio');
    textArea.at(0).simulate('change', { target: { value: 'sample' } });
    expect(textArea.at(0).text()).toEqual('sample');
  });

  it('Test InputTextArea in blur event with afterChange', () => {
    const input = wrapper.find('SampleForm').instance();
    input.setState({ afterChangeState: true });
    const textArea = wrapper.find('#bio');
    textArea.at(0).simulate('blur', { target: { value: ' Sample ' } });
    expect(textArea.at(0).text()).toEqual('Sample');
  });
  it('Test InputTextArea in blur event without afterChange', () => {
    const input = wrapper.find('SampleForm').instance();
    input.setState({ afterChangeState: false });
    const textArea = wrapper.find('#bio');
    textArea.at(0).simulate('blur', { target: { value: ' Sample ' } });
    expect(textArea.at(0).text()).toEqual('Sample');
  });
});
