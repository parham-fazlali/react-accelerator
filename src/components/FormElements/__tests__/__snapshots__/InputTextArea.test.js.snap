// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleForm /> on InputTextArea Test InputTextArea Change event with afterChange props 1`] = `ReactWrapper {}`;

exports[`<SampleForm /> on InputTextArea Test InputTextArea in blur event with afterChange 1`] = `ReactWrapper {}`;

exports[`<SampleForm /> on InputTextArea Test InputTextArea in blur event without afterChange 1`] = `ReactWrapper {}`;

exports[`<SampleForm /> on InputTextArea Test InputTextArea in change without afterChange props 1`] = `ReactWrapper {}`;

exports[`<SampleForm /> on InputTextArea Test InputTextArea with isFill validation error 1`] = `ReactWrapper {}`;
